buildlib/pr/ucx-py/ucx-py-test.yml

```diff
@@ -0,0 +1,90 @@
+parameters:
+  name: ucx-py test
+  demands: []
+
+jobs:
+  - job: ${{ parameters.name }}
+
+    pool:
+      name: MLNX
+      demands: ${{ parameters.demands }}
+
+    steps:
+      - checkout: self
+        fetchDepth: 100
+        clean: true
+        displayName: Checkout
+      - bash: |
+          set -x
+          source buildlib/az-helpers.sh
+          az_init_modules
+          try_load_cuda_env
+          set -eE
+          ./autogen.sh
+          ./contrib/configure-devel --prefix=$(Build.Repository.LocalPath)/install \
+            --enable-gtest=no --with-cuda=$have_cuda
+          make -j`nproc`
+          make install
+        displayName: Build UCX
+      - bash: |
+          set -x
+          source buildlib/az-helpers.sh
+          az_init_modules
+          try_load_cuda_env
+          set -eE
+
+          ifaces=`get_rdma_interfaces`
+          if [ -z "$ifaces" ]; then
+              azure_log_warning "No active RDMA interfaces on machine"
+              exit 0;
+          fi
+
+          # Set path and build parallel level
+          export PATH=/opt/conda/bin:$PATH
+
+          ################################################################################
+          # SETUP - Install conda packages and check environment
+          ################################################################################
+          gpuci_logger "Check environment"
+          env
+
+          gpuci_logger "Activate conda env"
+          . /opt/conda/etc/profile.d/conda.sh
+          conda activate ucx-py
+          gpuci_conda_retry install \
+              "pytest" "pytest-asyncio" \
+              "dask" "distributed" \
+              "cupy=9" "numba" \
+              "cython" "libhwloc"
+
+          gpuci_logger "Clone and Build UCX-Py"
+          git clone https://github.com/rapidsai/ucx-py
+          cd ucx-py
+          python setup.py build_ext --inplace
+          python -m pip install -e .
+
+          gpuci_logger "Check versions"
+          python --version
+          conda info
+          conda config --show-sources
+          conda list --show-channel-urls
+          ucx_info -v
+
+          ################################################################################
+          # TEST - Run UCX-Py tests and benchmarks
+          ################################################################################
+          for tls in "tcp,cuda_copy" "rc,tcp,cuda_copy" "all"; do
```

Comment:
```
for tls in "tcp,cuda" "ib,cuda" "all"
do
```

Reply:
Using `cuda` instead of `cuda_copy` is equivalent to `cuda_copy,cuda_ipc`, right? We would like to also test the non-CUDA IPC scenario, in the past we had different issues depending on transports being used, so the more we can test the better. Would `for tls in "tcp,cuda_copy" "tcp,cuda" "ib,cuda" "all"` be reasonable here? I intentionally didn't try adding `cuda_ipc` as I recall from a previous conversation a couple of months there was no NVLink support in CI.

Reply:
I see the point, maybe do something like:
```
test_tls="all" # All transports
test_tls+=" tcp,cuda_copy"  # Cuda without IPC
test_tls+=" tcp,cuda_copy,cuda_ipc"  # Cuda with IPC
if have_rdma_devices
then
    test_tls+=" rc,cuda_copy"
fi
...
```



```

Reply:
Good idea, done in https://github.com/openucx/ucx/pull/7412/commits/ecca05be22e6ea6888ddeba07c74b413568207d4 .